### PR TITLE
Set AppContext BaseDirectory

### PR DESCRIPTION
**Pull request by schellap**
@gkhanna79 PTAL

Will be ported to rel 1.0.0 as well once this goes through.

@piotrpMSFT @brthor

### CODE REVIEW

**Comment by schellap** [comment]
https://github.com/dotnet/coreclr/issues/2128

**Comment by brthor** [comment]
This will solve many issues we are seeing with the PkG and homebrew formulas on mac :shipit:

File: src/corehost/cli/hostpolicy.cpp [hunk 1]
```diff
@@ -106,6 +106,8 @@ int run(const arguments_t& args, const pal::string_t& clr_path)
         "AppDomainCompatSwitch",
         // TODO: pipe this from corehost.json
         "SERVER_GC",
+        // Workaround: CoreCLR doesn't resolve symlinks for AppContext.BaseDirectory
```
Comment (gkhanna79) [inline src/corehost/cli/hostpolicy.cpp]: FX, not CoreCLR.

**Comment by gkhanna79** [comment]
LGTM

File: src/corehost/cli/hostpolicy.cpp (hunk 1 above)
Comment (schellap) [inline src/corehost/cli/hostpolicy.cpp]: Will change, but mscorlib is considered FX?

Comment (gkhanna79) [inline src/corehost/cli/hostpolicy.cpp]: We should then call it out as "Mscorlib".

**Comment by piotrpMSFT** [comment]
@schellap why not PR into rel/1.0.0 directly? We merge all rel commits to master for you...

**Comment by schellap** [comment]
Okay! Thanks for the tip!

**Comment by piotrpMSFT** [comment]
I'll wait for the rel/1.0.0 PR then.

**Pull request closed without merging**